src/tap.rs

```diff
@@ -9,6 +9,10 @@
 
 use core::ops::Drop;
 
+pub fn diagnostic(msg: &str) {
+    serial_println!("# {}", msg);
+}
```

Comment:
I know this doesn't technically use the "count", but to me this could definitely be in the TestGroup structure. It removes the need for an extra import, it makes sense from the black-box perspective of other files that a diagnostic message would somehow be specific to a TestGroup even though technically it is not in this case. Finally, the difference between `tap::diagnostic("...")` and `tap.diagnostic("...")` is not much.

Reply:
I don't believe you can use the dot operator unless the method takes the struct as an argument in some form. So what we'd have to do for that is `tap::TestGroup::diagnostic(...)` which I originally tried but it seems silly.

Reply:
Oh, I meant it as an instance method, taking the struct as an argument. You're right that `tap::TestGroup::diagnostic` is silly. Technically, we don't need taking the struct as an argument, but I think it fits.